arch/arm/src/imx6/imx_gpio.c

```diff
@@ -542,16 +544,25 @@ int imx_config_gpio(gpio_pinset_t pinset)
         }
         break;
 
-      case GPIO_PERIPH:
-        {
-          /* Configure the pin as a peripheral */
-
-          ret = imx_gpio_configperiph(pinset);
-        }
-        break;
-
       default:
-        ret = -EINVAL;
+
+        /* Workaround for:
+         * chip/imx_gpio.c:545:7: error: case label does not reduce to an
+         *   integer constant
+         *    545 |       case GPIO_PERIPH:
+         *        |       ^~~~
+         */
+
+        if (pinset == GPIO_PERIPH)
```

Comment:
but GPIO_PERIPH is a const macro?

Reply:
Yes, `GPIO_PERIPH` is (2 << 30)

Reply:
this change is wired. How it can be reproduced?

Reply:
Use sabre-6quad:nsh, and enable CONFIG_MM_UBSAN and CONFIG_MM_UBSAN_ALL

Reply:
I will try it out. The first glance seems like this is because only bit 31 is set, but I'm not aware of such limitations in C.

Reply:
Normally this message is displayed when someone use a variable instead of a constant: https://www.includehelp.com/c-programs/case-label-does-not-reduce-to-an-integer-constant-error-in-c.aspx

This error shouldn't happen when using GPIO_PERIPH.

@no1wudi did you try replacing "case GPIO_PERIPH" with "case  (2 << 30)" just to confirm that the issue could be related to the signal bit?

Hmm, interesting test:

```
  const int c = 1;

  case c:
    printf("case c\n"); break;
```

clang accepts it, but gcc (11.3.0) raises an error:

```
testcase.c:12:9: error: case label does not reduce to an integer constant
   12 |         case c:     //no compile error for clang
      |         ^~~~
```

Reply:
@acassis Yes, i
 have tried (2<<30), and this issue also exists in riscv gcc.

Reply:
so it's gcc bug when UBSan is enabled?

Reply:
Yes, it should be a bug or
 limit with ubsan enabled.

Reply:
how about we ue 0x80000000?

Reply:
It works,  so should we change the definition of this macro?

Reply:
at least, it's better than the current change.

Reply:
Should we fill a bug with the GCC bug tracking system?

Reply:
I believe that it is not a single place in the code that will have such an issue. Also a question will `(2u << 30)` also generate an issue, or is it only a limitation of int?

Reply:
（2u << 30）will not report this issue, changed to unsigned definitions now.

Reply:
> Should we fill a bug with the GCC bug tracking system?

Yes, I think so.

> I believe that it is not a single place in the code that will have such an issue. Also a question will `(2u << 30)` also generate an issue, or is it only a limitation of int?

it's a problem of UBSan, I think. The origin code work well without UBSan enable.